{
  "uid" : "43950bf9368617f9",
  "name" : "Send OTP",
  "fullName" : "tests.test_auth.TestAuth#test_send_otp",
  "historyId" : "a7980f8fa59964950d3d067ba7162304",
  "time" : {
    "start" : 1754278939711,
    "stop" : 1754278939712,
    "duration" : 1
  },
  "description" : "Test send OTP endpoint.",
  "descriptionHtml" : "<p>Test send OTP endpoint.</p>\n",
  "status" : "broken",
  "statusMessage" : "requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?",
  "statusTrace" : "self = <test_auth.TestAuth object at 0x7fb162468830>\nemail = 'freevoodoo@yandex.com', expected = True\n\n    @allure.title(\"Send OTP\")\n    @pytest.mark.parametrize(\"email, expected\", [\n        pytest.param(Payloads.valid_email(), True, id=\"valid email\"),\n        pytest.param(Payloads.invalid_email(), False, id=\"invalid email\"),\n    ])\n    @pytest.mark.auth\n    def test_send_otp(self, email, expected):\n        \"\"\"Test send OTP endpoint.\"\"\"\n>       response = self.auth_api.send_otp(email=email, success=expected)\n\ntests/test_auth.py:37: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nservices/auth/api.py:34: in send_otp\n    response = self.client.post(\nclient/api_client.py:103: in post\n    response = self._request(\"post\", endpoint, json=json, data=data, headers=headers, files=files, success=success)\nclient/api_client.py:59: in _request\n    response = self.session.request(method=method, url=url, headers=headers, **kwargs)\n../local/lib/python3.12/site-packages/requests/sessions.py:575: in request\n    prep = self.prepare_request(req)\n../local/lib/python3.12/site-packages/requests/sessions.py:484: in prepare_request\n    p.prepare(\n../local/lib/python3.12/site-packages/requests/models.py:367: in prepare\n    self.prepare_url(url, params)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <PreparedRequest [POST]>, url = '/api/v1/auth/send-otp'\nparams = OrderedDict()\n\n    def prepare_url(self, url, params):\n        \"\"\"Prepares the given HTTP URL.\"\"\"\n        #: Accept objects that have string representations.\n        #: We're unable to blindly call unicode/str functions\n        #: as this will include the bytestring indicator (b'')\n        #: on python 3.x.\n        #: https://github.com/psf/requests/pull/2238\n        if isinstance(url, bytes):\n            url = url.decode(\"utf8\")\n        else:\n            url = str(url)\n    \n        # Remove leading whitespaces from url\n        url = url.lstrip()\n    \n        # Don't do any URL preparation for non-HTTP schemes like `mailto`,\n        # `data` etc to work around exceptions from `url_parse`, which\n        # handles RFC 3986 only.\n        if \":\" in url and not url.lower().startswith(\"http\"):\n            self.url = url\n            return\n    \n        # Support for unicode domain names and paths.\n        try:\n            scheme, auth, host, port, path, query, fragment = parse_url(url)\n        except LocationParseError as e:\n            raise InvalidURL(*e.args)\n    \n        if not scheme:\n>           raise MissingSchema(\n                f\"Invalid URL {url!r}: No scheme supplied. \"\n                f\"Perhaps you meant https://{url}?\"\n            )\nE           requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?\n\n../local/lib/python3.12/site-packages/requests/models.py:438: MissingSchema",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1754278939606,
      "stop" : 1754278939608,
      "duration" : 2
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "stepsCount" : 0,
    "hasContent" : false
  }, {
    "name" : "_xunit_setup_method_fixture_TestAuth",
    "time" : {
      "start" : 1754278939710,
      "stop" : 1754278939710,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "stepsCount" : 0,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "Test send OTP endpoint.",
    "status" : "broken",
    "statusMessage" : "requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?",
    "statusTrace" : "self = <test_auth.TestAuth object at 0x7fb162468830>\nemail = 'freevoodoo@yandex.com', expected = True\n\n    @allure.title(\"Send OTP\")\n    @pytest.mark.parametrize(\"email, expected\", [\n        pytest.param(Payloads.valid_email(), True, id=\"valid email\"),\n        pytest.param(Payloads.invalid_email(), False, id=\"invalid email\"),\n    ])\n    @pytest.mark.auth\n    def test_send_otp(self, email, expected):\n        \"\"\"Test send OTP endpoint.\"\"\"\n>       response = self.auth_api.send_otp(email=email, success=expected)\n\ntests/test_auth.py:37: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nservices/auth/api.py:34: in send_otp\n    response = self.client.post(\nclient/api_client.py:103: in post\n    response = self._request(\"post\", endpoint, json=json, data=data, headers=headers, files=files, success=success)\nclient/api_client.py:59: in _request\n    response = self.session.request(method=method, url=url, headers=headers, **kwargs)\n../local/lib/python3.12/site-packages/requests/sessions.py:575: in request\n    prep = self.prepare_request(req)\n../local/lib/python3.12/site-packages/requests/sessions.py:484: in prepare_request\n    p.prepare(\n../local/lib/python3.12/site-packages/requests/models.py:367: in prepare\n    self.prepare_url(url, params)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <PreparedRequest [POST]>, url = '/api/v1/auth/send-otp'\nparams = OrderedDict()\n\n    def prepare_url(self, url, params):\n        \"\"\"Prepares the given HTTP URL.\"\"\"\n        #: Accept objects that have string representations.\n        #: We're unable to blindly call unicode/str functions\n        #: as this will include the bytestring indicator (b'')\n        #: on python 3.x.\n        #: https://github.com/psf/requests/pull/2238\n        if isinstance(url, bytes):\n            url = url.decode(\"utf8\")\n        else:\n            url = str(url)\n    \n        # Remove leading whitespaces from url\n        url = url.lstrip()\n    \n        # Don't do any URL preparation for non-HTTP schemes like `mailto`,\n        # `data` etc to work around exceptions from `url_parse`, which\n        # handles RFC 3986 only.\n        if \":\" in url and not url.lower().startswith(\"http\"):\n            self.url = url\n            return\n    \n        # Support for unicode domain names and paths.\n        try:\n            scheme, auth, host, port, path, query, fragment = parse_url(url)\n        except LocationParseError as e:\n            raise InvalidURL(*e.args)\n    \n        if not scheme:\n>           raise MissingSchema(\n                f\"Invalid URL {url!r}: No scheme supplied. \"\n                f\"Perhaps you meant https://{url}?\"\n            )\nE           requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?\n\n../local/lib/python3.12/site-packages/requests/models.py:438: MissingSchema",
    "steps" : [ {
      "name" : "Send OTP",
      "time" : {
        "start" : 1754278939711,
        "stop" : 1754278939712,
        "duration" : 1
      },
      "status" : "broken",
      "statusMessage" : "requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?\n",
      "statusTrace" : "  File \"/usr/local/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/usr/workspace/services/auth/api.py\", line 34, in send_otp\n    response = self.client.post(\n               ^^^^^^^^^^^^^^^^^\n  File \"/usr/workspace/client/api_client.py\", line 103, in post\n    response = self._request(\"post\", endpoint, json=json, data=data, headers=headers, files=files, success=success)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/workspace/client/api_client.py\", line 59, in _request\n    response = self.session.request(method=method, url=url, headers=headers, **kwargs)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/requests/sessions.py\", line 575, in request\n    prep = self.prepare_request(req)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/requests/sessions.py\", line 484, in prepare_request\n    p.prepare(\n  File \"/usr/local/lib/python3.12/site-packages/requests/models.py\", line 367, in prepare\n    self.prepare_url(url, params)\n  File \"/usr/local/lib/python3.12/site-packages/requests/models.py\", line 438, in prepare_url\n    raise MissingSchema(\n",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "2442dd4f308b31eb",
        "name" : "request (json)",
        "source" : "2442dd4f308b31eb.json",
        "type" : "application/json",
        "size" : 40
      } ],
      "parameters" : [ {
        "name" : "email",
        "value" : "'freevoodoo@yandex.com'"
      }, {
        "name" : "success",
        "value" : "True"
      } ],
      "attachmentsCount" : 1,
      "shouldDisplayMessage" : true,
      "stepsCount" : 0,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "92f7ca3d3ea4f262",
      "name" : "log",
      "source" : "92f7ca3d3ea4f262.txt",
      "type" : "text/plain",
      "size" : 81
    } ],
    "parameters" : [ ],
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "stepsCount" : 1,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "_xunit_setup_method_fixture_TestAuth::0",
    "time" : {
      "start" : 1754278939742,
      "stop" : 1754278939742,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "stepsCount" : 0,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "story",
    "value" : "Authentication"
  }, {
    "name" : "epic",
    "value" : "Auth"
  }, {
    "name" : "tag",
    "value" : "auth"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_auth"
  }, {
    "name" : "subSuite",
    "value" : "TestAuth"
  }, {
    "name" : "host",
    "value" : "fe4ea36ad96b"
  }, {
    "name" : "thread",
    "value" : "1-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_auth"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "email",
    "value" : "'freevoodoo@yandex.com'"
  }, {
    "name" : "expected",
    "value" : "True"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 3,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 3
      },
      "items" : [ {
        "uid" : "ab94f6994c7203ac",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?",
        "time" : {
          "start" : 1754278243206,
          "stop" : 1754278243208,
          "duration" : 2
        }
      }, {
        "uid" : "270c89be091b5ba0",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.MissingSchema: Invalid URL '/api/v1/auth/send-otp': No scheme supplied. Perhaps you meant https:///api/v1/auth/send-otp?",
        "time" : {
          "start" : 1754277805345,
          "stop" : 1754277805346,
          "duration" : 1
        }
      } ]
    },
    "tags" : [ "auth" ]
  },
  "source" : "43950bf9368617f9.json",
  "parameterValues" : [ "'freevoodoo@yandex.com'", "True" ]
}